common/kernel/nvme_ctrlr_mgmt.c

```diff
@@ -459,6 +459,63 @@ DumpRegs(struct NvmeCtrlr *ctrlr, struct usr_io *uio)
 
    return vmk_CopyToUser(uio->addr, ctrlr->regs, length);
 }
+/**
+ * PCIe registers read/write
+ *
+ * @param [in] ctrlr controller instance
+ * @param [in] uio pointer to the uio structure
+ *
+ * @return VMK_OK if completes successfully;
+ * otherwise return error code
+ */
+static VMK_ReturnStatus
+PciRegsOP(struct NvmeCtrlr *ctrlr, struct usr_io *uio)
+{
+    struct pcie_regs_rw regs;
+    vmk_PCIDevice dev = ctrlr->ctrlOsResources.pciDevice;
+    VMK_ReturnStatus vmkStatus = VMK_OK;
+
+    VMK_ASSERT(uio->length == sizeof(regs));
+    vmk_CopyFromUser((vmk_VA)&regs, uio->addr, sizeof(regs));
+    VMK_ASSERT(regs.size == 4);
```

Comment:
Please remove ASSERT because we should not allow the userworld application panic the kernel. For the failure of sanity check, I think we could just return failure status to notify the application.